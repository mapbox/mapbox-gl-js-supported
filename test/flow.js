'use strict';
const fs = require('fs');
const test = require('tap').test;
const spawn = require('child_process').spawnSync;

test('Flow types', (t) => {
    const expected = fs.readFileSync(`${__dirname}/flow.expected.txt`, 'utf8');
    const result = spawn('npx', [
        'flow',
        '--color=never',
        `${__dirname}/flow.input.js.flow`
    ], {encoding: 'utf8'});
    const actual = result.stdout;
    if (process.env.UPDATE) {
        fs.writeFileSync(`${__dirname}/flow.expected.txt`, actual);
    }
    t.deepEqual(actual, expected);
    t.end();
});


