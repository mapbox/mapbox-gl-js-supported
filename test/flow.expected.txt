Error ------------------------------------------------------------------------------------ test/flow.input.js.flow:11:20

Cannot assign `isSupported()` to `r3` because boolean [1] is incompatible with string [2].

   test/flow.input.js.flow:11:20
   11| const r3: string = isSupported();
                          ^^^^^^^^^^^^^

References:
   index.js.flow:2:83
    2| declare type isSupported = (options?: {failIfMajorPerformanceCaveat: boolean}) => boolean;
                                                                                         ^^^^^^^ [1]
   test/flow.input.js.flow:11:11
   11| const r3: string = isSupported();
                 ^^^^^^ [2]


Error ------------------------------------------------------------------------------------ test/flow.input.js.flow:12:33

Cannot call `isSupported` with object literal bound to `options` because property `failIfMajorPerformanceCaveat` is
missing in object literal [1] but exists in object type [2].

   test/flow.input.js.flow:12:33
   12| const r4: boolean = isSupported({});
                                       ^^ [1]

References:
   index.js.flow:2:39
    2| declare type isSupported = (options?: {failIfMajorPerformanceCaveat: boolean}) => boolean;
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [2]



Found 2 errors
